server/methods.js

```diff
@@ -216,14 +216,28 @@ Meteor.methods({
   */
   flickrImageSearchList: function(query, option, page) {
     var items, nextPage;
+    var photo_id = '';
     check(query, String);
+
+    if ((query.indexOf('flickr.com') !==-1) && (query.indexOf('/photos/') !==-1)){
+      //search photo: flickr.com/photos/{user-id}/{photo-id}/in/photolist-{search-info}
+      //individual photo:  flickr.com/photos/{user-id}/{photo-id}
+      var split = _.chain(query.split('/')).compact().value();
+      photo_id =  (split[split.indexOf('photos') +2]).match(/[\d]*/)[0];
+    } else if ((query.indexOf('flic.kr') !==-1) && (query.indexOf('/p/') !==-1)){
+      //short url: https://flic.kr/p/{base58-photo-id}
+      photo_id = _.chain(query.split('/')).compact().last().value().match(/[\d\w]*/)[0];
+    }
+
     page = page || 1;  // flickr starts from 1
     this.unblock();
 
-    var url = "https://api.flickr.com/services/rest/?&method=flickr.photos.search";
+    var path = photo_id ? 'flickr.photos.getInfo' : 'flickr.photos.search';
+    var url = "https://api.flickr.com/services/rest/?&method=" + path;
 
     var requestParams = {
       tags: query.replace(' ', ','),
+      photo_id: photo_id,
```

Comment:
Each query has its own set of parameters, but it "worked anyway" when you pile all the requests together because the parameters get ignored. But I see how this could make things confusing if flickr changes their api in the future.